segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/ChunkHandle.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.storage.chunklayer;
+
+import lombok.Data;
+import lombok.NonNull;
+
+/**
+ * Handle to a chunk.
+ */
+@Data
+final public class ChunkHandle {
```

Comment:
I apologize,
I am trying to understand better the final API that we are defining.

It looks like that this ChunkHandle is only a wrapper for a _chunkName_.
It looks mostly useless.

As you previously said, it would be really useful that concrete implementations subclass this class and enrich it with custom properties.

Reply:
I agree - In fact it is useless, and I'm now inclined to remove the concept of ChunkHandle . 

The problem with extensible ChunkHandle is that now to create read/write handle instances in the code , we have to  add openRead and openWrite as new apis on ChunkStorageProvider. (so that we create an instance of correct concrete type). 

It initially seemed useful to have ability to put implementation specific data in ChunkHandle.
However then problem is that when you have two separate ChunkHandle instances pointing to the same underlying object, the data stored in them diverges and can have unintended side effects .
In this case most likely implementation choice is to keep single copy of internal state in some other data structure and handle merely points to that data. So having extensible ChunkHandle isn't really helping here.

Given that ChunkHandle creates more problem than it solves - I am not sure having it is useful.

Reply:
For now keeping the ChunkHandle.